Add contributors + Add blog 01 (#11)
NucleoidMC/nucleoid.xyz: content/blog/01_mapmaking.md

+++
title = "Map Making"
date = 2021-05-01

extra.image = "assets/images/blog/mapmaking.png"
+++

> Map Making 101

Reviewer: This should probably be a title- also, an alternative title suggestion: "How We Make Our Maps"?

> Map Making 101! How else do you think we made our minigames?

Reviewer 2: ```suggestion
Map Making 101
```

> When a builder decides that they want to make a map, they first of all, head to our build server (Whitelisted for obvious reasons) found at <code>build.nucleoid.xyz</code>

Reviewer 2: ```suggestion
When a builder decides that they want to make a map, they first of all, head to our (whitelisted) build server found at <code>build.nucleoid.xyz</code>
```

They choose a game and create a map using a handy command.

<br>

> <img class="header_image" src="https://iili.io/Kg34kl.jpg">

Reviewer: Images would rather be done through markdown if possible- also, can we move the images into the static content on the site so that it doesn't have to go off-site?

> One of our builders is creating a [KotH (King of the Hill)](https://github.com/NucleoidMC/koth) map using [Plasmid](https://github.com/NucleoidMC/plasmid). Plasmid is a framework that allows our games to run, maps loaded and much much more. Its considered one of the main reasons why NucleoidMC isn't using bukkit or commonly bungeecord.

Reviewer 2: ```suggestion
One of our builders is creating a [KotH (King of the Hill)](https://github.com/NucleoidMC/koth) map using [Plasmid](https://github.com/NucleoidMC/plasmid). Plasmid is our custom game library which allows us to create minigames, and maps for those minigames, with ease.
```

> Plasmid allows our builders to access a <code>/map</code> command. The map command allows the swift creation of maps, editing, and exporting for use in production. 

Reviewer: Code blocks should also be possible through markdown

Author: The markdown codeblocks aren't rendererd. I think its a zola bug.

Reviewer: It was missing in the CSS- it should be fine now to use markdown.

> Plasmid allows our builders to access a <code>/map</code> command. The map command allows the swift creation of maps, editing, and exporting for use in production. 

Reviewer: Can also describe that when creating a map, it opens a new dimension. Can possibly mention fantasy as a library being involved- but not sure if that would be interesting. Depends who our audience is- and tbh, I have no idea! :D

Author: Our audience is the people interested in the techy shit 😄

> Each map requires regions (Unless the game doesn't require any set regions) which can be configured using the <code>/map region</code> command.

Reviewer: I think it might be nice to describe what a region means and the importance of it- the details of what the command is specifically is not super important here to the casual reader I imagine.

Author: This blog gets on the technical side, i plan to make a blog on how the server builds the mods on restart/start which wont be for the faintest of readers.

Reviewer: Yeah- though the specific of the name of the command is not so interesting technically as opposed to the concept of what a region means.

For KotH, we require two vital regions:

> <br><br>

Reviewer: Linebreaks through markdown too (can add double space after line to make a linebreak)

Author: Zola, again, doesn't render these linebreaks.

- Spawn - Where the players will spawn after being knocked off or at the start of the game.

- Throne - Where the "top of the hill" is considered.

<br>

For showcase purposes, our builder has set the regions to visible:

<br>
<img class="header_image" src="https://iili.io/Kg3ZBe.png">

"But what do the colors mean?" you might say;
<br><br>
- Red - The bounding box, anything built outside this red cube will not be saved.

> - Blue - Regions, as you can see, the lower region is the spawn and the topmost region is the throne.

Reviewer 2: ```suggestion
- Blue - Regions, the lower region is the spawn and the topmost region is the throne.
```

<br>

By the looks of it, our builder has finished their map. 

> All they need to do is run <code>/map export</code> and their map will be ready to download at [the file system found here.](https://build.nucleoid.xyz/plasmid/export)

Reviewer: Might be good to add somewhere a link to the Plasmid wiki [creating a map documentation](https://github.com/NucleoidMC/plasmid/wiki/Creating-a-Map) as an out-of-interest extra reading thing.

+++
title = "Map Making"
date = 2021-05-01

extra.image = "assets/images/blog/mapmaking.png"
+++

> How we make our maps:

Reviewer: ```suggestion
# How We Make Our Maps
```

When a builder decides that they want to make a map, they first of all, head to our (whitelisted) build server found at <code>build.nucleoid.xyz</code>

They choose a game and create a map using a handy command.

<br>

> <img class="header_image" src="assets/images/blog/Kg34kl.jpg">

Reviewer: I believe this should work through markdown: `![Image Description](assets/images/Kg34kl.jpg)`

Author: Markdown results in the image being too big and overflow off the page. Hence the `header_image` class

Reviewer: This should be resolved in the CSS then- an image within a blog overflowing the page is never desirable :D

Author: This is a blog post PR, not a CSS bug fix spree lol. I'll make a seperate PR with the fixes later on.
